src/java/htsjdk/samtools/SAMUtils.java

```diff
@@ -558,10 +561,20 @@ public static void makeReadUnmapped(final SAMRecord rec) {
         rec.setMappingQuality(SAMRecord.NO_MAPPING_QUALITY);
         rec.setInferredInsertSize(0);
         rec.setNotPrimaryAlignmentFlag(false);
+        rec.setSupplementaryAlignmentFlag(false);
         rec.setProperPairFlag(false);
         rec.setReadUnmappedFlag(true);
     }
 
+    /**
+     * Strip mapping information from a SAMRecord, but preserve it in the 'O' tags.
+     */
+    public static void makeReadUnmappedWithOriginalTags(final SAMRecord rec) {
+        rec.setAttribute(SAMTag.OP.name(), rec.getAlignmentStart());
```

Comment:
You should only overwrite the 'O' tags if they aren't already set. this will keep their status as "originalX" rather than "previousX"...